File: bot-api/java/src/test/java/test_utils/BotInfoBuilder.java
package test_utils;

import dev.robocode.tankroyale.botapi.BotInfo;
import dev.robocode.tankroyale.botapi.InitialPosition;

import java.util.Collection;
import java.util.List;

public final class BotInfoBuilder {

    private String name;
    private String version;
    private List<String> authors;
    private String description;
    private String homepage;
    private List<String> countryCodes;
    private Collection<String> gameTypes;
    private String platform;
    private String programmingLang;
    private InitialPosition initialPosition;

    public BotInfoBuilder(BotInfo botInfo) {
        name = botInfo.getName();
        version = botInfo.getVersion();
        authors = botInfo.getAuthors();
        description = botInfo.getDescription();
        homepage = botInfo.getHomepage();
        countryCodes = botInfo.getCountryCodes();
        gameTypes = botInfo.getGameTypes();
        platform = botInfo.getPlatform();
        programmingLang = botInfo.getProgrammingLang();
        initialPosition = botInfo.getInitialPosition();
    }

    public BotInfo build() {
        return new BotInfo(name, version, authors, description, homepage, countryCodes, gameTypes, platform,
                programmingLang, initialPosition);
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public void setAuthors(List<String> authors) {
        this.authors = authors;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setHomepage(String homepage) {
        this.homepage = homepage;
    }

    public void setCountryCodes(List<String> countryCodes) {
        this.countryCodes = countryCodes;
    }

    public void setGameTypes(List<String> gameTypes) {
        this.gameTypes = gameTypes;
    }

    public void setPlatform(String platform) {
        this.platform = platform;
    }

    public void setProgrammingLang(String programmingLang) {
        this.programmingLang = programmingLang;
    }

    public void setInitialPosition(InitialPosition initialPosition) {
        this.initialPosition = initialPosition;
    }
}

File: bot-api/dotnet/Robocode.TankRoyale.BotApi.Tests/src/test_utils/BotInfoBuilder.cs
using System.Collections.Generic;

namespace Robocode.TankRoyale.BotApi.Tests.Test_utils;

public class BotInfoBuilder
{
    public BotInfoBuilder(BotInfo botInfo)
    {
        Name = botInfo.Name;
        Version = botInfo.Version;
        Authors = botInfo.Authors;
        Description = botInfo.Description;
        Homepage = botInfo.Homepage;
        CountryCodes = botInfo.CountryCodes;
        GameTypes = botInfo.GameTypes;
        Platform = botInfo.Platform;
        ProgrammingLang = botInfo.ProgrammingLang;
        InitialPosition = botInfo.InitialPosition;
    }

    public BotInfo Build()
    {
        return new BotInfo(Name, Version, Authors, Description, Homepage, CountryCodes, GameTypes, Platform,
            ProgrammingLang, InitialPosition);
    }

    public string Name { get; set; }

    public string Version { get; set; }

    public IEnumerable<string> Authors { get; set; }

    public string Description { get; set; }

    public string Homepage { get; set; }

    public IEnumerable<string> CountryCodes { get; set; }

    public IEnumerable<string> GameTypes { get; set; }

    public string Platform { get; set; }

    public string ProgrammingLang { get; set; }

    public InitialPosition InitialPosition { get; set; }
}